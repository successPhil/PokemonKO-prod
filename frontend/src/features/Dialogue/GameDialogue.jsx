import Pokeball from "../../styles/Pokeball";
import TypingEffect from "../../styles/Typewriter";

export default function GameDialogue({text}) {

  return (
    <div className='gameboy-dialogue-container'>
      <div className='gameboy-dialogue-box'>
        <div className='dialogue-content'>
        {text && <TypingEffect text={text} /> }
        </div>
      </div>
      <div className="top-left">
        <Pokeball />
      </div>
      <div className="top-right">
        <Pokeball />
      </div>
      <div className="bottom-left">
        <Pokeball />
      </div>
      <div className="bottom-right">
        <Pokeball />
      </div>
    </div>
  );
}

