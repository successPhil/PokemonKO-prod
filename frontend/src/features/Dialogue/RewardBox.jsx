import TypingEffect from "../../styles/Typewriter"
export default function RewardBox({text}) {
    return(
        <>
        <div className='reward-box-container'>
            <div className='gameboy-menu-box'>
                <div className='reward-box-content'> {text && <TypingEffect text={text} /> }</div>
      </div>
    </div>
      
        </>
    )
}