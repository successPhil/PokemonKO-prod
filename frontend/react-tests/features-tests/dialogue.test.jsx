import GameDialogue from '../../src/features/Dialogue/GameDialogue'
import RewardBox from '../../src/features/Dialogue/RewardBox'
import {expect, test} from 'vitest'


test("Game Dialogue renders without crashing", ()=>{
    const gameDialogue = <GameDialogue/>
    expect(gameDialogue).toMatchSnapshot()

    const rewardBox = <RewardBox />
    expect(rewardBox).toMatchSnapshot()

})