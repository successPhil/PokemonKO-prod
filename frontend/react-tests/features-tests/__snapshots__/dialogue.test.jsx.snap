// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Dialogue renders without crashing 1`] = `<GameDialogue />`;

exports[`Game Dialogue renders without crashing 2`] = `<RewardBox />`;
